### PR TITLE
[MXNET-1083] Add the example to demonstrate the inference workflow using C++ API.

### PR DESCRIPTION
**Pull request by leleamol**
## Description ##
The PR includes an example that demonstrates the inference workflow using C++ API.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [y] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [y] Changes are complete (i.e. I finished coding on this PR)
- [y] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [y] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)
The example demonstrates how to load the pre-trained model and associated parameter files.
The model and parameter files can be specified as command line arguments.

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by roywei** [comment]
@leleamol Thanks for the contribution!
@mxnet-label-bot [pr-awaiting-review]

**Comment by ankkhedia** [comment]
@nswamy Could you please help review the PR.

File: cpp-package/example/inference/inception_inference.cpp [hunk 1]
```diff
@@ -0,0 +1,380 @@
...
+ *
+ * This class encapsulates the functionality to load the model, process input image and run the forward pass.
+ */
+
+class Predictor {
+ public:
+    Predictor() {}
+    Predictor(const std::string model_json,
+              const std::string model_params,
+              const Shape& input_shape,
```
[the remark this answers is not in the document]
Comment (leleamol) [reply cpp-package/example/inference/inception_inference.cpp]: @nswamy This Predictor() is specific to this example. Hence, I would like to keep it simple. I agree, for a generic Predictory(), this can be improved. But it is beyond scope of this example.

@cjolivier01 This is a oversight. I should have passed string as a reference. Will fix it.

File: cpp-package/example/inference/inception_inference.cpp [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+    output_labels.push_back(lemma);
+  }
+  fi.close();
+}
+
+
+/*
+ * The following function loads the input image.
+ */
+void Predictor::LoadInputImage(const std::string& image_file) {
```
[the remark this answers is not in the document]
Comment (leleamol) [reply cpp-package/example/inference/inception_inference.cpp]: This is a demo example and not meant for testing mutli-threaded execution. The Predictor class and it's functions are written only for this example.

File: cpp-package/example/inference/inception_inference.cpp [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+
+/*
+ * The following function normalizes input image data by substracting the
+ * mean data.
+ */
+void Predictor::NormalizeInput(const std::string& mean_image_file) {
+  LG << "Normalizing image using " << mean_image_file;
+  mean_img = NDArray(input_shape, global_ctx, false);
+  mean_img.SyncCopyFromCPU(
+        NDArray::LoadToMap(mean_image_file)["mean_img"].GetData(),
```
[the remark this answers is not in the document]
Comment (leleamol) [reply cpp-package/example/inference/inception_inference.cpp]: This can be optimized. Will fix it. No need to load for every inference.

**Comment by leleamol** [comment]
@nswamy @cjolivier01 
Thanks for the review.  This is an example to demonstrate a simple inference workflow using pre-trained network. The Predictor class and it's method are not a written to be consumed as a generic API. We definitely can write such an API but it will increase the complexity of this example.

**Comment by leleamol** [comment]
Opened a new PR #13294
